wolfcrypt/src/wc_xmss_impl.c

```diff
@@ -2689,6 +2689,9 @@ static void wc_xmss_bds_state_load(const XmssState* state, byte* sk,
     /* Skip past standard SK = idx || wots_sk || SK_PRF || root || SEED; */
     sk += params->idx_len + 4 * n;
 
+    if (2 * (int)params->d - 1 < 0)
+        return WC_FAILURE;
```

Comment:
d is not going to be 0 (it is the number of subtrees).
Return with no error would be fine unless ctidy needs it to bail later.

Reply:
Decided to leave the error return there, because the later clause,
```
    if (wots_sigs != NULL) {
        *wots_sigs = sk;
    }
```
would be silently missed if we return success before entering the loop.  Don't want to have to think about implications of that, and as you say, `d > 0` unless something has gone very wrong.